docs/src/devdocs/FEValues.md

```diff
@@ -26,3 +26,17 @@ Ferrite.FunctionValues
   the jacobian, $\partial M/\partial x$, is calculated. 
 * In the second part of `reinit!`, The `MappingValues`, containing sufficient information for the current quadrature point, is given to `apply_mapping!`. 
   This allows the shape values and gradients stored in `FunctionValues`, to be mapped to the current cell by calling `apply_mapping!`.
+
+## Custom FEValues
+Custom FEValues, `fe_v`, should normally implement the `reinit!` method.
+Additionally, for normal functionality the `getnquadpoints` should be implemented.
+Note that asking for the `n`th quadrature point must be inside array bounds if 
+`1<=n<:getnquadpoints(fe_v)`
+(`checkquadpoint` can, alternatively, be dispatched to check that `n` is inbounds.)
+
+Supporting `function_value`, `function_gradient`, `function_symmetric_gradient`, `function_divergence`, and `function_curl`,
+requires implementing `getnbasefunctions`, `shape_value`, and `shape_gradient`. 
+Note that asking for the `i`th shape value or gradient must be inside array bounds if `1<=i<:getnbasefunctions(fe_v)`
+
+Supporting `spatial_coordinate` requires implementing `getngeobasefunctions` and `geometric_value`.
```

Comment:
Same as above.

Reply:
Same answer as above, but not shown in the `SimpleCellValues` example.